src/tidy.zig

```diff
@@ -9,280 +10,516 @@ const Ast = std.zig.Ast;
+            }
+        }
     }
+}
 
-    return null;
+test tidy_banned {
+    try check_tidy_file(
+        \\banned.zig
+    ,
+        \\//FIXME: use copy_disjoint:
+        \\@memcpy(foo, bar)
+    ,
+        snap(@src(),
+            \\banned.zig:2: error: @memcpy( is banned, use stdx.copy_disjoint
+            \\banned.zig:1: error: leftover FIXME, remove before merge
+            \\
+        ),
+    );
 }
 
-fn tidy_long_line(file: SourceFile) !?u32 {
-    if (std.mem.endsWith(u8, file.path, "low_level_hash_vectors.zig")) return null;
+fn tidy_lines(file: SourceFile, errors: *Errors) void {
+    if (std.mem.endsWith(u8, file.path, "low_level_hash_vectors.zig")) return;
+
     var line_iterator = mem.splitScalar(u8, file.text, '\n');
     var line_index: u32 = 0;
     while (line_iterator.next()) |line| : (line_index += 1) {
-        const line_length = try std.unicode.utf8CountCodepoints(line);
-        if (line_length > 100) {
-            if (has_link(line)) continue;
-
-            // Journal recovery table
-            if (std.mem.indexOf(u8, line, "Case.init(") != null) continue;
-
-            // For multiline strings, we care that the _result_ fits 100 characters,
-            // but we don't mind indentation in the source.
-            if (parse_multiline_string(line)) |string_value| {
-                const string_value_length = try std.unicode.utf8CountCodepoints(string_value);
-                if (string_value_length <= 100) continue;
-
-                if (std.mem.endsWith(u8, file.path, "state_machine_tests.zig") and
-                    (std.mem.startsWith(u8, string_value, " account A") or
-                        std.mem.startsWith(u8, string_value, " transfer T") or
-                        std.mem.startsWith(u8, string_value, " transfer   ")))
-                {
-                    // Table tests from state_machine.zig. They are intentionally wide.
-                    continue;
-                }
+        tidy_line(file, line, line_index, errors);
+    }
+}
 
-                // vsr.zig's Checkpoint ops diagram.
-                if (std.mem.endsWith(u8, file.path, "vsr.zig") and
-                    std.mem.startsWith(u8, string_value, "OPS: ")) continue;
+fn tidy_line(file: SourceFile, line: []const u8, line_index: usize, errors: *Errors) void {
+    const line_length = tidy_line_length(line);
+    if (line_length <= 100) return;
 
-                // trace.zig's JSON snapshot test.
-                if (std.mem.endsWith(u8, file.path, "trace.zig") and
-                    std.mem.startsWith(u8, string_value, "{\"pid\":1,\"tid\":")) continue;
+    if (tidy_line_link(line)) return;
 
-                // AMQP JSON snapshot test.
-                if (std.mem.endsWith(u8, file.path, "cdc/runner.zig") and
-                    std.mem.startsWith(u8, string_value, "{\"timestamp\":")) continue;
-            }
+    // Journal recovery table
+    if (std.mem.indexOf(u8, line, "Case.init(") != null) return;
```

Comment:
Again the principle of no silencing lints. If something _really_ wants to silence the lint, the silencing part goes directly to the lint definition itself.